glib/src/unichar.rs

```diff
@@ -0,0 +1,238 @@
+    #[doc(alias = "unichar_break_type")]
+    #[inline]
+    fn break_type(self) -> UnicodeBreakType {
+        unsafe { from_glib(ffi::g_unichar_break_type(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_get_script")]
+    #[doc(alias = "unichar_get_script")]
+    #[inline]
+    fn script(self) -> UnicodeScript {
+        unsafe { from_glib(ffi::g_unichar_get_script(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_combining_class")]
+    #[doc(alias = "unichar_combining_class")]
+    #[inline]
+    fn combining_class(self) -> u8 {
+        // UAX #44 § 5.7.4: The character property invariants regarding Canonical_Combining_Class
+        //                  guarantee that [...] all values used will be in the range 0..254.
+        // So this cast is fine
+        unsafe { ffi::g_unichar_combining_class(self.into()) as u8 }
+    }
+
+    #[doc(alias = "g_unichar_ismark")]
+    #[doc(alias = "unichar_ismark")]
+    #[inline]
+    fn is_mark(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_ismark(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_isgraph")]
+    #[doc(alias = "unichar_isgraph")]
+    #[inline]
+    fn is_graphical(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_isgraph(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_ispunct")]
+    #[doc(alias = "unichar_ispunct")]
+    #[inline]
+    fn is_punctuation(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_ispunct(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_istitle")]
+    #[doc(alias = "unichar_istitle")]
+    #[inline]
+    fn is_titlecase(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_istitle(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_isdefined")]
+    #[doc(alias = "unichar_isdefined")]
+    #[inline]
+    fn is_defined(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_isdefined(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_iswide")]
+    #[doc(alias = "unichar_iswide")]
+    #[inline]
+    fn is_wide(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_iswide(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_iswide_cjk")]
+    #[doc(alias = "unichar_iswide_cjk")]
+    #[inline]
+    fn is_wide_cjk(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_iswide_cjk(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_iszerowidth")]
+    #[doc(alias = "unichar_iszerowidth")]
+    #[inline]
+    fn is_zero_width(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_iszerowidth(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_totitle")]
+    #[doc(alias = "unichar_totitle")]
+    #[inline]
+    fn to_titlecase(self) -> char {
+        Self::from_unichar(unsafe { ffi::g_unichar_totitle(self.into()) })
+    }
+
+    #[doc(alias = "g_unichar_get_mirror_char")]
+    #[doc(alias = "unichar_get_mirror_char")]
+    #[inline]
+    fn mirror_char(self) -> Option<char> {
+        let mut mirrored = MaybeUninit::uninit();
+        let res = unsafe {
```

Comment:
The whole function body should really be in an `unsafe` here. The three steps of creating uninitialized memory, calling the function and only assuming the memory is initialized when the function succeeds all belong together and are the same "safety promise". Just try writing a `SAFETY` comment for each of the individual `unsafe` blocks and you'll see what I mean (you'll have to refer to the other `unsafe` blocks!).